### PR TITLE
Bicaws 2154 filter by defendantname

### PR DESCRIPTION
**Pull request by joe-fol**
Add logic to query with a case-insensitive match.
Update listCourtCases to allow search by defendant name. 
PR for front end to follow

### CODE REVIEW

File: src/types/QueryParams.ts [hunk 1]
```diff
@@ -4,5 +4,6 @@ export type QueryParams = {
   orderBy?: string
```
Comment (emadkaramad) [inline src/types/QueryParams.ts:4]: I would assume this file is only for the cases list page. Should we prefix the filename with the page name to be more specific?

**Review by emadkaramad — APPROVED** [review]

File: src/useCases/listCourtCases.ts [hunk 2]
```diff
@@ -15,6 +15,12 @@ const listCourtCases = async (connection: DataSource, queryParams: QueryParams):
     )
     .limit(queryParams.limit)
 
+  if (queryParams.defendantName) {
+    query.where("lower(courtCase.defendantName) like '%' || :name || '%'", {
```
Comment (emadkaramad) [inline src/useCases/listCourtCases.ts:19]: For case-insensitive search in Postgres, it seems we can use [ILIKE](https://www.postgresql.org/docs/9.0/functions-matching.html) keyword. So we don't need to use `lower()` function.

Comment (C-gyorfi) [reply src/useCases/listCourtCases.ts]: Refactored in [b22d455](https://github.com/ministryofjustice/bichard7-next-ui/pull/24/commits/b22d455b17a38c3e89b99d3442fa71a1e35c2587)

File: src/types/QueryParams.ts (hunk 1 above)
Comment (C-gyorfi) [reply src/types/QueryParams.ts:4]: Refactored in [b22d455](https://github.com/ministryofjustice/bichard7-next-ui/pull/24/commits/b22d455b17a38c3e89b99d3442fa71a1e35c2587)

**Review by C-gyorfi — APPROVED** [review]

**Pull request merged**